### PR TITLE
fix: change to log.fail

### PR DESCRIPTION
**Pull request by gideonairex**

### CODE REVIEW

**Comment by gideonairex** [comment]
@yamii review please

File: index.js [hunk 1]
```diff
@@ -58,10 +58,11 @@ function setConfigs ( options ) {
 	 * options might not only be rabbit
 	 */
 
-	config.rabbit = options;
```
Comment (yamii) [inline index.js]: Please add another if to hasMultipleOptions for timeout

File: index.js [hunk 2]
```diff
@@ -58,10 +58,11 @@ function setConfigs ( options ) {
...
+	config.rabbit  = options;
+	config.timeout.setOptions( options.timeout );
 	if ( hasMultipleOptions( options ) ) {
-		config.rabbit = options.rabbit;
-		config.logger = options.logger;
+		config.rabbit  = options.rabbit;
+		config.logger  = options.logger;
```
Comment (yamii) [inline index.js]: set hasOptions for timeout inside if

File: lib/req-res/responder.js [hunk 3]
```diff
@@ -49,11 +49,12 @@ Responder.prototype.consume = function ( callback ) {
...
-			'messageType' : that.messageType,
-			'respond'     : respond,
-			'message'     : message,
-			'emitter'     : that
+			'log'            : that.logger,
+			'messageType'    : that.messageType,
+			'respond'        : respond,
+			'message'        : message,
+			'emitter'        : that,
+			'timeoutHandler' : config.timeout
```
Comment (yamii) [inline lib/req-res/responder.js]: Why not require config in `status.js`?

File: lib/req-res/status.js [hunk 4]
```diff
@@ -2,53 +2,53 @@
...
 					'status' : 'success',
 					'data'   : data
 				} );
 				setImmediate( function () {
 					log.silly( 'success', {
-						'data'        : data,
-						'messageType' : options.messageType,
-						'message'     : options.message
-					} );
+							'data'        : data,
```
Comment (yamii) [inline lib/req-res/status.js]: fix indention

File: lib/util/timeout.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+/* eslint consistent-return:0 */
+function TimeoutHandler () {
+	this.disable = true;
+}
+
+TimeoutHandler.prototype.setOptions = function ( options ) {
+	if ( options ) {
+		this.disable = false;
```
Comment (yamii) [inline lib/util/timeout.js]: fix alignment

File: lib/util/timeout.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		respond( {
+			'status' : 'fail',
+			'data'   : 'Request timeout'
+		} );
+		emitter.emit( 'timeout', {
+			'messageType' : options.messageType,
+			'message'     : options.message
+		} );
+		setImmediate( function () {
+			log.warn( 'timeout', {
```
Comment (openam) [inline lib/util/timeout.js]: Should this be a warn or error? It seems like if this happens we have bigger problems.

Comment (gideonairex) [inline lib/util/timeout.js]: Makes sense to make it as error.

**Comment by gideonairex** [comment]
@yamii  @openam please review :)

**Pull request merged**